authenticate-users/index.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+# Copyright 2019 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+// "getallheaders" works for NGINX headers in PHP 7.3, but not in PHP 7.2 or below.
+require_once __DIR__ . '/getallheaders.php';
+
+# [START getting_started_auth_all]
+require_once __DIR__ . '/vendor/autoload.php';
+
+# [START getting_started_auth_certs]
+/**
+ * Returns a dictionary of current Google public key certificates for
+ * validating Google-signed JWTs. Since these change rarely, the result
+ * is cached on first request for faster subsequent responses.
+ */
+function certs() : string
+{
+    $client = new Guzzle\Http\Client();
+    $response = $client->get(
+        'https://www.gstatic.com/iap/verify/public_key-jwk'
+    )->send();
+    return $response->getBody();
+}
+# [END getting_started_auth_certs]
+
+# [START getting_started_auth_metadata]
+/**
+ * Returns a string with the project metadata value for the item_name.
+ * See https://cloud.google.com/compute/docs/storing-retrieving-metadata for
+ * possible item_name values.
+ */
+function get_metadata($itemName) : string
+{
+    $client = new Guzzle\Http\Client();
+
+    $endpoint = 'http://metadata.google.internal';
+    $path = '/computeMetadata/v1/project/' . $itemName;
+    $response = $client->get(
```

Comment:
I would understand wanting to keep this independent of relying on too many libraries, but if you'd like we do offer support for communicating with the metadata server in `google/cloud-core`: https://github.com/googleapis/google-cloud-php/blob/master/Core/src/Compute/Metadata.php

Reply:
I'm not super against to including it, but as it's a simple HTTP request, and there's quite a bit in the `google/core` package, I feel better leaving it out.

Reply:
So I swapped it out for the metadata class and it looks a lot better, so we'll go with that!